app/models/users/user.rb

```diff
@@ -1,6 +1,28 @@
 class User < ActiveRecord::Base
   devise :database_authenticatable, :trackable,
-         :validatable#, :omniauthable
+         :validatable, :omniauthable, omniauth_providers: [:google_oauth2]
+  has_and_belongs_to_many :oauth_credentials
   belongs_to :school
   belongs_to :grove
+
+  def self.from_omniauth(access_token)
+    data = access_token.info
```

Comment:
never mind, i think this is fine here. just return `nil` if there is not user record found.